nox/_parametrize.py

```diff
@@ -13,9 +13,59 @@
 # limitations under the License.
 
 import functools
+import itertools
 
 
-def parametrize_decorator(arg_names, arg_values_list):
+class Param:
+    """A class that encapsulates a single set of parameters to a parametrized
+    session."""
```

Comment:
1. You know my PEP257 feelings on these one-liners that span multiple lines
1. Worth documents the constructor arguments?

Reply:
2. Done. :)